src/ServiceBasedMerchant/ServiceBasedMerchantState.php

```diff
@@ -0,0 +1,240 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\ServiceBasedMerchant;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Registerable;
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\OptionsAwareInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\OptionsAwareTrait;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\OptionsInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\TransientsAwareInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\TransientsAwareTrait;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\TransientsInterface;
+use WC_Product;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class ServiceBasedMerchantState
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\ServiceBasedMerchant
+ */
+class ServiceBasedMerchantState implements Service, Registerable, OptionsAwareInterface, TransientsAwareInterface {
+
+	use OptionsAwareTrait;
+	use TransientsAwareTrait;
+
+	/**
+	 * Batch size for product queries.
+	 *
+	 * @var int
+	 */
+	private const BATCH_SIZE = 100;
+
+	/**
+	 * Cache expiration time in seconds (1 hour).
+	 *
+	 * @var int
+	 */
+	private const CACHE_EXPIRATION = HOUR_IN_SECONDS;
+
+	/**
+	 * Register hooks to clear cache when products are updated.
+	 */
+	public function register(): void {
+		// Clear cache when products are added or updated.
+		// Note: woocommerce_update_product passes (int $product_id, WC_Product $product)
+		add_action( 'woocommerce_new_product', [ $this, 'clear_cache_on_product_change' ], 10 );
+		add_action( 'woocommerce_update_product', [ $this, 'clear_cache_on_product_change' ], 10, 2 );
+		add_action( 'woocommerce_new_product_variation', [ $this, 'clear_cache_on_product_change' ], 10 );
+		add_action( 'woocommerce_update_product_variation', [ $this, 'clear_cache_on_product_change' ], 10, 2 );
+
+		// Clear cache when products are deleted or trashed.
+		add_action( 'wp_trash_post', [ $this, 'maybe_clear_cache_on_post_change' ], 10, 2 );
+		add_action( 'before_delete_post', [ $this, 'maybe_clear_cache_on_post_change' ], 10 );
+		add_action( 'deleted_post', [ $this, 'maybe_clear_cache_on_post_change' ], 10 );
+
+		// Clear cache when products are restored from trash.
+		add_action( 'untrashed_post', [ $this, 'maybe_clear_cache_on_post_change' ], 10, 2 );
+
+		// Clear cache when WooCommerce clears product transients.
+		add_action( 'woocommerce_delete_product_transients', [ $this, 'clear_cache_on_product_change' ], 10 );
```

Comment:
@jamesmorrison we only need to determine if a store is a service based merchant _once_ during the initial onboarding stage. Therefore we shouldn't be tracking changes to products and updating the value again.

Can we remove these hooks and the cache clear method.